federated_aws_cli/sts_conn.py

```diff
@@ -25,22 +25,30 @@ def get_credentials(bearer_token, role_arn):
         local_username = pwd.getpwuid(os.getuid())[0]
         role_session_name = 'federated-aws-cli-{}'.format(local_username)
         sts_url = "https://sts.amazonaws.com/"
-        parameters = {
-            'Action': 'AssumeRoleWithWebIdentity',
-            'RoleArn': role_arn,
-            'RoleSessionName': role_session_name,
-            'WebIdentityToken': bearer_token,
-            'Version': '2011-06-15'
-        }
-
-        # Call the STS API
-        resp = requests.get(url=sts_url, params=parameters)
-        if resp.status_code != requests.codes.ok:
-            logger.error('AWS STS Call failed {} : {}'.format(resp.status_code, resp.text))
-            return None
-
-        logger.debug('STS Call Response headers : {}'.format(resp.headers))
-        logger.debug('STS Call Response : {}'.format(resp.text))
+        duration_seconds = [3600, 43200]  # 1 hour, 12 hours
```

Comment:
I find this construction to be a little un-Pythonic. I think it would be lot clearer as:

```python
for duration_seconds in [43200, 3600]:
    do_stuff()

    if resp.status_code == requests.codes.ok:
        break
else:
    error_out()
```

That would clear up the awkward `while` loop and things happening in the reverse order due to the `.pop()` call.

Reply:
Good idea, I made this change in a4fd30ad4e2dbb6e0695e37a82575ca97006af85